### PR TITLE
Update spatial mesh level of detail values, translate in observer

### PR DESCRIPTION
**Pull request by davidkline-ms**
This is a minor breaking change (values in the SpatialMeshLevelOfDetail enum) to eliminate direct triangles per cubic meter encoding in the enum.

This change adds a Medium LOD as well as a lookup method in the windows mesh observer.

Verified by changing the LOD in the observer profile and deploying / running on HoloLens.

### CODE REVIEW

File: Assets/MixedRealityToolkit.Providers/WindowsMixedReality/WindowsMixedRealitySpatialMeshObserver.cs [hunk 1]
```diff
@@ -236,14 +236,47 @@ public SpatialAwarenessMeshLevelOfDetail LevelOfDetail
...
+                case SpatialAwarenessMeshLevelOfDetail.Coarse:
+                    triangleDensity = 0;
+                    break;
+
+                case SpatialAwarenessMeshLevelOfDetail.Medium:
+                    triangleDensity = 400;
+                    break;
+
+                case SpatialAwarenessMeshLevelOfDetail.Fine:
+                    triangleDensity = 200;
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit.Providers/WindowsMixedReality/WindowsMixedRealitySpatialMeshObserver.cs]: Is `Fine` intended to be less dense than `Medium`?

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit.Providers/WindowsMixedReality/WindowsMixedRealitySpatialMeshObserver.cs]: good catch. thanks!

**Pull request merged**